src/client/texturesource.cpp

```diff
@@ -179,22 +185,30 @@ TextureSource::TextureSource()
 	m_textureinfo_cache.emplace_back(TextureInfo{""});
 	m_name_to_id[""] = 0;
 
-	// Cache some settings
-	// Note: Since this is only done once, the game must be restarted
-	// for these settings to take effect.
+	for (auto *name : {"mip_map", "trilinear_filter", "bilinear_filter",
+		"anisotropic_filter"}) {
+		g_settings->registerChangedCallback(name,
+			&settingChangedCallback, this);
```

Comment:
looks like these change callbacks aren't of any use since the `[applyfiltersformesh` implementation in `imagesource.cpp` still uses cached values?

Reply:
texture and image source get re-created when joining a game so this is pointless actually.

Reply:
except if we want to support changes in-game (clientmap.cpp might already deal with them properly)